express-postgres/server.ts

```diff
@@ -0,0 +1,84 @@
+import express from 'express';
+import { ClientError, errorMiddleware } from './lib/index.js';
+import pg from 'pg';
+
+const app = express();
+
+app.use(errorMiddleware);
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/pagila',
+  ssl: {
+    // Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+
+app.get('/api/films', async (req, res, next) => {
+  try {
+    const sql = `
+      select
+        "title"
+      from "films"
+      order by  "replacementCost" DESC;
+    `;
+    const result = await db.query(sql);
+    const actors = result.rows;
+    res.send(actors);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get('/api/films/:filmId', async (req, res, next) => {
+  try {
+    const { filmId } = req.params;
+    if (filmId === undefined) {
+      throw new ClientError(400, 'filmId is required');
+    }
+    const sql = `
+      select
+        "title",
+        "filmId
+      from "films"
+    `;
+    const params = [filmId];
+    const result = await db.query(sql, params);
```

Comment:
This should throw an exception because you are providing a parameter (filmId) but there is not a `$1` in the SQL. You need a where clause, `where "filmId" = $1`.